packages/widget-v2/src/modals/TokenAndChainSelectorModal/TokenAndChainSelectorModal.tsx

```diff
@@ -207,6 +207,14 @@ export const TokenAndChainSelectorModal = createModal(
           return -1;
         }
 
+        if (assetB.asset.originChainID === assetB.chainID) {
+          return 1;
+        }
+
+        if (assetA.asset.originChainID === assetA.chainID) {
+          return -1;
+        }
+
```

Comment:
This is the sorting change, everything else is for consolidating the fetch balance back to one request, ever since the refactor I believe there's no longer much of a benefit to fetching balances seperately